roles/rac-lsnr-firewall/tasks/main.yml

```diff
@@ -45,3 +45,24 @@
   when: create_listener
   tags: lsnr-firewall
 
+- name: Allow local networks for RAC
+  firewalld:
+    zone: public
+    rich_rule: rule family=ipv4 source address="{{ item.value.ipv4.network }}/{{ item.value.ipv4.netmask}}" accept
+    permanent: yes
+    immediate: yes
+    state: enabled
+  become: yes
+  with_items:
+    - "{{ ansible_facts | dict2items | selectattr('value.ipv4.network', 'defined') |list }}"
+  tags: lsnr-firewall
```

Comment:
Some parts of the config for the existing firewalld on line 17 looks useful:
- Don't run if disable_firewall is set
- Search for an indication of success in firewalld output (requiring a variable to be registered to store the output)

Reply:
#### 1: For the first point:
... of skipping the task when `disable_firewall=True`, added this `when` clause and tested successfully that the block gets executed.

Also moved the call to the role to its own block as follows with its own tag:
```
- hosts: dbasm
  serial: 1
  tasks:
    - name: rac-gi-install | defaults from common
      include_vars:
        dir: roles/common/defaults
    - name: rac-gi-install | open firewall
      include_role:
        name: rac-lsnr-firewall
      when:
        - cluster_type == "RAC"
  tags: lsnr-firewall
```

This will enable the particular role / block to be called surgically using `-- "--tags lsnr-firewall" ` without invoking other roles as follows:
```
nohup ./install-oracle.sh --ora-swlib-bucket gs://bmaas-testing-oracle-software \
--instance-ssh-user ansible --instance-ssh-key ~/.ssh/id_rsa --backup-dest /u01/backups \
--ora-swlib-path /u01/oracle_install --ora-version 19 --ora-swlib-type gcs --cluster-type RAC \
--cluster-config mntrl_cluster.json --install-sw  -- "--tags lsnr-firewall"  > ~/050521-mntrl-instlln/bms-toolkit/logs/mntrl.test_firewall_PR_run_7   2>&1   &
```
#### 2: For the second point:
... of adding a success check based on the o/p received:

This is the entire output first run when firewalld module changes the  firewall rules (internal):
https://paste.googleplex.com/5907082843783168

This is the entire o/p during subsequent idempotent runs (internal):
https://paste.googleplex.com/4835607269343232

Based on above, couldn't check for successful changes to firewall rules.
Instead of checking for a successful checks, will borrow existing code block to check for errors during the firewalld run as follows:
```
...
  register: firewall_op
  failed_when:
    - "'firewall is not currently running' not in firewall_op.msg"
    - "'Permanent and Non-Permanent(immediate) operation' not in firewall_op.msg"
  when: not disable_firewall|bool
  tags: lsnr-firewall
```

Reply:
Looks good!